glprogs/stages/depth/depth.params.glsl

```diff
@@ -0,0 +1,14 @@
+#pragma tdm_define "MAX_SHADER_PARAMS"
+
+struct ShaderParams {
+    mat4 modelViewMatrix;
+	mat4 textureMatrix;
+	vec4 color;
+	vec4 scissor;
+	vec2 alphaTest;
```

Comment:
UBOs have very strict alignment rules, so I can't put a single float here, because the entire struct would then not adhere to those rules. (Or actually, here in the shader I could, but it would add implicit padding which you'd need to account for the mirrored struct in the C++ code, which I think would be even more confusing.)

But on second thought, it'd probably be better to use two floats and name the second one `padding` or something similar.

Reply:
Padding sounds better. Some people also use names like `_` or `_1`, `_2`, etc. It would immediately show the reason for all of this. The problem of different layout will hunt us all the time now, so reader has know know about it.

2D vector with unused sounds more confusing. Reminds ARB shaders with their "everything is vec4" rule =)

Reply:
Yep. I've already put in place a STATIC_ASSERT that will tell you if your params struct is not a multiple of 16, but this doesn't catch all the mistakes you can do. Still better than nothing, I think.